### PR TITLE
Single source of truth for depot and other safes

### PR DESCRIPTION
**Pull request by lukemelia**
Likely an alternative to #2031

### CODE REVIEW

File: packages/web-client/app/resources/depot.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    await taskFor(this.viewSafesTask).perform(this.args.named.walletAddress);
+  }
+
+  get value() {
+    let safes = taskFor(this.viewSafesTask).lastSuccessful?.value;
+    let depotSafes = (safes || []).filter(
+      (safe: Safe) => safe.type === 'depot'
+    ) as DepotSafe[];
+
+    let depot = depotSafes[depotSafes.length - 1] ?? null;
```
Comment (backspace) [inline packages/web-client/app/resources/depot.ts:35]: Is it concerning at all if there’s more than one depot?

File: packages/web-client/app/utils/web3-strategies/layer2-chain.ts [hunk 2]
```diff
@@ -230,15 +226,17 @@ export default abstract class Layer2ChainWeb3Strategy
   }
 
   async refreshBalances() {
-    return taskFor(this.fetchDepotTask).perform();
+    return taskFor(this.viewSafesTask).perform();
```
Comment (backspace) [inline packages/web-client/app/utils/web3-strategies/layer2-chain.ts:229]: Maybe current instances of `safes.fetch` can be replaced with a call to `refreshBalances`? To make that the standard interface.

**Review by backspace — APPROVED** [review]
I look forward to using this 🎉

File: packages/web-client/app/resources/depot.ts (hunk 1 above)
Comment (lukemelia) [reply packages/web-client/app/resources/depot.ts:35]: Yes, but I am going to trust the rest of the system to ensure that does not happen.

File: packages/web-client/app/utils/web3-strategies/layer2-chain.ts (hunk 2 above)
Comment (lukemelia) [reply packages/web-client/app/utils/web3-strategies/layer2-chain.ts:229]: Good idea.

**Pull request merged**